appvalidate/partner_secrets.go

```diff
@@ -0,0 +1,59 @@
+package appvalidate
+
+import (
+	"context"
+	"errors"
+	"fmt"
+)
+
+var (
+	ErrUnsupportedPartner        = errors.New("unsupported partner")
+	ErrPartnerUnitialized        = errors.New("partner not initialized")
+	ErrInvalidPartnerCredentials = errors.New("invalid partner credentials")
+	ErrEmptySecretsConfig        = errors.New("empty secrets config")
+	ErrInvalidPartnerPayload     = errors.New("invalid partner payload")
+)
+
+type PartnerSecrets struct {
+	cs *CoastalSecrets
+	ps *PalmTreeSecrets
+}
+
+// NewPartnerSecrets creates a new Secrets wrapper with all supported
```

Comment:
The callers should be checking for nil, and if found, they should just continue or fail. Using a half-baked secrets isn't going to work anyway. 

As far as I can tell, the callers do just that:

https://github.com/tidepool-org/platform/blob/master/auth/service/api/v1/appvalidate.go#L153

I don't think we should be returning a half-initialized PartnerSecrets. 

If the callers really need to have something non-nil (but as far as I can tell, they don't), then a dummy implementation should be returned that just returns dummy responses and maybe logs to remind people that it's a dummy that's in use.